src/adapter/src/coord/ddl.rs

```diff
@@ -525,6 +529,9 @@ impl Coordinator {
             if update_http_config {
                 self.update_http_config();
             }
+            if update_enable_persist_txn_tables {
+                self.update_txns_config().await;
+            }
```

Comment:
I think you're betting off removing this and updating the following method: https://github.com/MaterializeInc/materialize/blob/5b3586ffa4396e08be157bb984d22af07f5b40c5/src/adapter/src/catalog.rs#L3195-L3205

You would have to add a method to `Transaction`, if one doesn't already exist, then something the following to the end:
```Rust
        if name == ENABLE_PERSIST_TXN_TABLES {
            tx.set_enable_persist_txn_tables(state.system_configuration.enable_persist_txn_tables());
        }
```

That way both the config and the system var values will be updated in the same transaction. With the current approach it seems like the two values could end up drifting apart if we get a failure after the catalog transaction commits but before this is run.

Also you'd be able to remove `set_enable_persist_txn_tables` from `DurableCatalogState`.

Reply:
Yeah, that's much better, thanks!